goal_src/levels/sunken/sun-iris-door.gc

```diff
@@ -5,3 +5,600 @@
+      )
+     )
+    )
+   )
+  :code
+  (behavior ()
+   (sound-play-by-name
+    (static-sound-name "irisdoor2")
+    (new-sound-id)
+    1024
+    0
+    0
+    1
+    #t
+    )
+   (let ((a0-2 (-> self skel root-channel 0)))
+    (set!
+     (-> a0-2 param 0)
+     (the float (+ (-> a0-2 frame-group data 0 length) -1))
+     )
+    (set! (-> a0-2 param 1) 1.0)
+    (joint-control-channel-group!
+     a0-2
+     (the-as art-joint-anim #f)
+     num-func-seek!
+     )
+    )
+   (until (ja-done? 0)
+    (suspend)
+    (let ((a0-3 (-> self skel root-channel 0)))
+     (set!
+      (-> a0-3 param 0)
+      (the float (+ (-> a0-3 frame-group data 0 length) -1))
+      )
+     (set! (-> a0-3 param 1) 1.0)
+     (joint-control-channel-group-eval!
+      a0-3
+      (the-as art-joint-anim #f)
+      num-func-seek!
+      )
+     )
+    )
+   (ja-post)
+   (go sun-iris-door-open)
+   (none)
+   )
+  :post
+  (behavior ()
+   (cond
+    ((-> self move-to?)
+     (set! (-> self move-to?) #f)
+     (TODO-RENAME-30 (-> self root-override) (-> self move-to-pos))
+     (quaternion-copy! (-> self root-override quat) (-> self move-to-quat))
+     (ja-post)
+     )
+    (else
+     (transform-post)
+     )
+    )
+   (none)
+   )
+  )
+
+(defmethod should-close? sun-iris-door ((obj sun-iris-door))
+  (let ((f30-0 1228800.0))
+   0.0
+   (let ((f0-7 (cond
+                ((-> obj directional-proximity?)
+                 (let ((s5-0 (new 'stack-no-clear 'vector)))
+                  (when *target*
+                   (vector-! s5-0 (target-pos 0) (-> obj root-override trans))
+                   (set! (-> s5-0 y) 0.0)
+                   (set! f30-0 (fabs (vector-dot s5-0 (-> obj outward-vec))))
+                   )
+                  (vector-! s5-0 (camera-pos) (-> obj root-override trans))
+                  (set! (-> s5-0 y) 0.0)
+                  (fabs (vector-dot s5-0 (-> obj outward-vec)))
+                  )
+                 )
+                (else
+                 (if *target*
+                  (set!
+                   f30-0
+                   (vector-vector-xz-distance
+                    (-> obj root-override trans)
+                    (target-pos 0)
+                    )
+                   )
+                  )
+                 (vector-vector-xz-distance
+                  (-> obj root-override trans)
+                  (camera-pos)
+                  )
+                 )
+                )
+          )
+         )
+    (when (and (>= f30-0 (-> obj close-dist)) (>= f0-7 (-> obj close-dist)))
+     (cond
+      ((and *target* (-> obj directional-proximity?))
+       (let ((s4-6 (new 'stack-no-clear 'vector))
+             (s5-3 (new 'stack-no-clear 'vector))
+             )
+        (vector-! s4-6 (target-pos 0) (-> obj root-override trans))
+        (set! (-> s4-6 y) 0.0)
+        (vector-! s5-3 (camera-pos) (-> obj root-override trans))
+        (set! (-> s5-3 y) 0.0)
+        (case (>= (vector-dot (-> obj outward-vec) s4-6) 0.0) 
```

Comment:
this is weird, I think it's the first case where the decompiler rewrote something as a `case` that probably wasn't originally a case.